### DISCUSSION TITLE
cacheFlags: true not updating flags

### DISCUSSION THREAD

**Discussion by BastiR9** [Flagsmith/flagsmith#424] [Q&A]
Hi all,
i started to experiment with flagsmith and have some trouble. I like the idea by caching my flags, but i don't understand why my flags will not be updated. 
Situation: 
1) Start App
2) get initial flags
3) store them in cache
4) change flags 
5) restart app
6) use flags in cache not new ones

Expectation:
1)Start App
2)get initial flags
3) store them in cache
4)change flags
5)restart app
6)use new flags and overwrite cache

Is there some very long waiting time coming with cacheFlag: true between flag updates? Did not found info in docs.
Thanks a lot

**Comment by dabeeeenster** [comment]
Is this in a particular SDK? Are you able to share some code?

**Reply by BastiR9** [reply]
I just use the "code example 2: Initializing your project" inside my dashboard BUT add the cacheFlags command. If i delete the command the flags will be updating immediately. 

`import flagsmith from "flagsmith"; //Add this line if you're using flagsmith via npm

flagsmith.init({
    environmentID:"xxxx",
    cacheFlags: true,
    onChange: (oldFlags, params) => { // Occurs whenever flags are changed
        // Determines if the update came from the server or local cached storage
        const { isFromServer } = params; 

        // Check for a feature
        if (flagsmith.hasFeature("my_cool_feature")) {
            myCoolFeature();
        }

        // Or, use the value of a feature
        const banner_size = flagsmith.getValue("banner_size");

        // Check whether value has changed
        const banner_sizeOld = oldFlags["banner_size"] 
        && oldFlags["banner_size"].value;
        
        if (banner_size !== banner_sizeOld) {
            // Value has changed, do something here
        }
    }
});`
